dms/models/directory.py

```diff
@@ -64,18 +65,35 @@ class DmsDirectory(models.Model):
         ondelete="restrict",
         auto_join=True,
         store=True,
+        compute_sudo=True,
     )
-
     parent_id = fields.Many2one(
         comodel_name="dms.directory",
-        domain="[('permission_create', '=', True)]",
         string="Parent Directory",
+        domain="[('permission_create', '=', True)]",
         ondelete="restrict",
-        auto_join=True,
+        prefetch=False,
```

Comment:
Why this? I think it requires an explanation as comment on code.

Reply:
Yes, it's because having access to a directory doesn't necessarily grant access to its parent, so while prefetching this you could get an unnecessary access error. I'll add the comment, it's really not obvious.